_sources/structs/mixed_up_code.rst

```diff
@@ -0,0 +1,236 @@
+the function - these lines should not be selected. Make sure to place the blocks at the right indentation levels!
+
+.. parsonsprob:: structs_mixed_up_code_1
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the struct definition of ``Song``. 
+    The Song structure will have the instance variables ``string title``, 
+    ``string artist``, ``string album``, and ``int year`` in that order. 
+    Put the necessary blocks of code in the correct order.
+    -----
+    struct Song {
+    =====
+    struct song { #paired
+    =====
+        string title;
+    =====
+        string artist;
+    =====
+        string album;
+    =====
+        int year;
+    =====
+        string year;  #distractor
+    =====
+    };
+    =====
+    } #distractor
+
+.. parsonsprob:: structs_mixed_up_code_2
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``printSong`` function. ``printSong``
+    takes a ``Song`` as a parameter and prints out the instance variables
+    in the following format: ``"title" by artist (album, year)``. Put the necessary blocks of 
+    code in the correct order.
+    -----
+    void printSong (Song &s) {
+    =====
+        cout << "\"" << s.title << "\" by " << s.artist;
+    =====
+        cout << " (" << s.album << ", " << s.year << ")" << endl;
+    =====
+        cout << title << artist << album << year;  #distractor
+    =====
+        cout << "\"" << title << "\" by " << artist;  #distractor
+    =====
+    }
+
+.. parsonsprob:: structs_mixed_up_code_3
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the struct definition of ``Unicorn``. 
+    The ``Unicorn`` structure will have the instance variables ``name``, 
+    ``age``, ``hornLength``, ``hairColor``, and ``isSparkly`` in that order. A Unicorn's
+    ``hornLength`` is measured to the nearest tenth of a unit and ``isSparkly`` is a boolean.
+    Put the necessary blocks of code in the correct order.
+    -----
+    struct Unicorn {
+    =====
+    Struct Unicorn {  #distractor
+    =====
+        string name;
+    =====
+        int age;
+    =====
+        double hornLength;
+    =====
+        string hairColor;
+    =====
+        bool isSparkly;
+    =====
+        int hornLength;  #distractor
+    =====
+    };
+    =====
+    } #distractor
+
+.. parsonsprob:: structs_mixed_up_code_4_re
+    :numbered: left
+    :adaptive:
+
+    Let's write the code for the ``convertToHumanAge`` function. ``convertToHumanAge``
+    takes a ``Unicorn`` as a parameter and returns the equivalent human age.
+    If a unicorn ``isSparkly``, then its equivalent human age is three times its ``age`` in unicorn years
+    plus the ``hornLength``. If a unicorn is not sparkly, then its equivalent human age is
+    four times its ``age`` in unicorn years plus twice the ``hornLength``.
+    Put the necessary blocks of code in the correct order.
```

Comment:
Specify in in the instructions that you must use an else statement (this could be solved without using the else statement, but then you get it wrong)